apps/rush/src/start.ts

```diff
@@ -35,44 +36,68 @@ else if (!semver.satisfies(nodeVersion, '^6.9.0')
 import * as path from 'path';
 import { JsonFile, IPackageJson } from '@microsoft/node-core-library';
 
-import {
-  RushConfiguration,
-  Rush
-} from '@microsoft/rush-lib';
-import { Utilities } from '@microsoft/rush-lib/lib/utilities/Utilities';
+import { Rush, EnvironmentVariableNames } from '@microsoft/rush-lib';
 
 import { MinimalRushConfiguration } from './MinimalRushConfiguration';
 import { RushVersionSelector } from './RushVersionSelector';
 
-const RUSH_PURGE_OPTION_NAME: string = 'purge';
+// Load the configuration
+const configuration: MinimalRushConfiguration | undefined = MinimalRushConfiguration.loadFromDefaultLocation();
+const currentPackageJson: IPackageJson = JsonFile.load(path.join(__dirname, '..', 'package.json'));
 
-if (process.argv[2] === RUSH_PURGE_OPTION_NAME) {
-  const rushDirectory: string = path.join(RushConfiguration.getHomeDirectory(), '.rush');
-  console.log(`Deleting ${rushDirectory} directory...`);
-  Utilities.dangerouslyDeletePath(rushDirectory);
-  console.log('done');
-} else {
-  // Load the configuration
-  const configuration: MinimalRushConfiguration | undefined = MinimalRushConfiguration.loadFromDefaultLocation();
-  const currentPackageJson: IPackageJson = JsonFile.load(path.join(__dirname, '..', 'package.json'));
-
-  // If we're inside a repo folder, and it's requesting a different version, then use the RushVersionManager to
-  // install it
-  if (configuration && configuration.rushVersion !== currentPackageJson.version) {
-    const versionSelector: RushVersionSelector = new RushVersionSelector(
-      configuration.homeFolder,
-      currentPackageJson.version
-    );
-    versionSelector.ensureRushVersionInstalled(configuration.rushVersion)
-      .catch((error: Error) => {
-        console.log(colors.red('Error: ' + error.message));
-      });
-  } else {
-    // Otherwise invoke the rush-lib that came with this rush package
-    const isManaged: boolean = !!configuration && configuration.rushVersion === currentPackageJson.version;
-    Rush.launch(
-      currentPackageJson.version,
-      isManaged // Rush is "managed" if its version and configuration are dictated by a repo's rush.json
-    );
+let rushVersionToLoad: string | undefined = undefined;
+
+const previewVersion: string | undefined = process.env[EnvironmentVariableNames.RUSH_PREVIEW_VERSION];
+
+if (previewVersion) {
+  if (!semver.valid(previewVersion, false)) {
+    console.error(colors.red(`Invalid value for RUSH_PREVIEW_VERSION: "${previewVersion}"`));
+    process.exit(1);
```

Comment:
>process.exit(1); [](start = 4, length = 16)

Can you put an explicit return?

Reply:
Returns "never"

---
In reply to: [187754990](https://github.com/Microsoft/web-build-tools/pull/653#discussion_r187754990) [](ancestors = 187754990)